### DISCUSSION TITLE
How to deal with large amount of insert logs in minio?

### DISCUSSION THREAD

**Discussion by peixin-lin** [milvus-io/milvus#16168] [Q&A and General discussion] [answered]
Our Milvus server crashed after deployed for one day. 
The reason is that a large amount of insert logs (~50G after 9h) in minio caused the disk full. 
Is there a way to minimized the log files saved? 
Or is it save to just delete these log files day by day (I am not sure will they be used by Milvus service)?

**Comment by yah01** [comment, marked as answer]
> Our Milvus server crashed after deployed for one day. 
> The reason is that a large amount of insert logs (~50G after 9h) in minio caused the disk full. 
> Is there a way to minimized the log files saved? 
> Or is it save to just delete these log files day by day (I am not sure will they be used by Milvus service)?
> 

Hi peixin, The binlogs in minio are what the milvus service relies on, they may be only compacted (to remove duplicates and entities been deleted), but not deleted entirely.
We did enable compression for binlogs, and it makes the data size be about 3x less than before. You can try the milvus at master branch, or wait for a release version that includes the compression feature.

**Reply by xiaofan-luan** [reply]
Do we enable parquet compression automatically?

**Reply by yah01** [reply]
> Do we enable parquet compression automatically?

Yes, the compression is all done by Arrow itself (which writes the parquet).

**Comment by xiaofan-luan** [comment]
～ minio is where the data persistence. One way to reduce amount of data is to enable compression, another way is to expire data by time, expire functionality is a feature working in progress

**Reply by wilsonlimaneto** [reply]
Hello all!! Did the expire funcionality get the chance go go into any of the releases? Thanks

**Reply by harpreetpaul** [reply]
Hi, is he expiry functionality now available with minio data persistence.

**Reply by xiaofan-luan** [reply]
If collection is dropped or compacted, it should be gced in 24 hours.
If not please upgrade to latest version and check if it helps

**Comment by xiaofan-luan** [comment]
> o go

Using TTL is probably the feature you are looking for.
